zio-http/jvm/src/test/scala/zio/http/endpoint/RoundtripSpec.scala

```diff
@@ -551,16 +551,32 @@ object RoundtripSpec extends ZIOHttpSpec {
             assert(r.isFailure)(isTrue) // We expect it to fail but complete
           }
       },
-      test("Override default CodecConfig") {
-        val api = Endpoint(GET / "test").out[Outs]
+      test("default CodecConfig preserves empty collections in response") {
+        val api = Endpoint(GET / "test").out[Ints]
         testEndpointCustomRequestZIO(
-          api.implement(_ => ZIO.succeed(Outs(Nil))).toRoutes @@ CodecConfig.withConfig(
-            CodecConfig(ignoreEmptyCollections = false),
-          ),
+          api.implement(_ => ZIO.succeed(Ints(Nil))).toRoutes,
           Request.get("/test"),
           response => response.body.asString.map(s => assertTrue(s == """{"ints":[]}""")),
         )
       },
+      test("default CodecConfig accepts missing collections in request") {
+        val api = Endpoint(POST / "test").in[Ints].out[Int]
+        testEndpointCustomRequestZIO(
+          api.implement(ints => ZIO.succeed(ints.ints.length)).toRoutes,
+          Request.post("/test", Body.fromString("""{}""")),
+          response => response.body.asString.map(s => assertTrue(s == "0")),
+        )
+      },
```

Comment:
To ensure the duality mentioned in the comment below

https://github.com/zio/zio-http/issues/3186#issuecomment-2465995247
> I too would expect empty collections to be represented in responses (but missing collections to be accepted in requests -- notice the duality).